ompi/mca/coll/base/coll_base_bcast.c

```diff
@@ -811,7 +812,7 @@ int ompi_coll_base_bcast_intra_scatter_allgather(
         if (vrank & mask) {
             int parent = (rank - mask + comm_size) % comm_size;
             /* Compute an upper bound on recv block size */
-            recv_count = count - vrank * scatter_count;
+            recv_count = (count > vrank * scatter_count) ? (count - vrank * scatter_count) : 0;
             if (recv_count <= 0) {
```

Comment:
Should this line be changed as well to

```suggestion
            if (recv_count == 0) {
```
?

Reply:
fixed